homeassistant/components/guardian/binary_sensor.py

```diff
@@ -121,22 +110,19 @@ def __init__(
         self,
         entry: ConfigEntry,
         coordinator: DataUpdateCoordinator,
-        kind: str,
-        name: str,
-        device_class: str | None,
-        icon: str | None,
+        description: BinarySensorEntityDescription,
     ) -> None:
         """Initialize."""
-        super().__init__(entry, coordinator, kind, name, device_class, icon)
+        super().__init__(entry, coordinator, description)
 
         self._attr_is_on = True
 
     @callback
     def _async_update_from_latest_data(self) -> None:
         """Update the entity."""
-        if self._kind == SENSOR_KIND_LEAK_DETECTED:
+        if self._description.key == SENSOR_KIND_LEAK_DETECTED:
```

Comment:
This should be `self.entity_description.key` instead.

Furthermore, I would recommend to assign it to a temporary variable first. Don't quite know if it's really faster, but certainly a bit easier to read. Something like this:
```py
sensor_type = self.entity_description.key
if sensor_type == ...:
    ...
elif sensor_type == ...:
    ...
```

Reply:
I don't like single-use temporary variables floating around just for moderate readability. I'll add a property to `GuardianEntity` that holds this info.